### PR TITLE
[MAJOR] Created project financials page.

### LINKED ISSUE #203: As a user, I want to manage the project financials
For this story to be accepted:
- Have the features listed above
- UI mockups
- Tests (frontend/backend)
- Diagrams (rest diagram/class diagram) (if needed)

### PR DESCRIPTION
**Pull request by kevin-ghannoum**
closes #203

### CODE REVIEW

File: backend/src/main/routes/TransactionRoute.ts [hunk 1]
```diff
@@ -118,6 +121,82 @@ export default class TransactionRoute extends CommonRoutesConfig {
         }
       });
 
+    this.getApp()
+      .route(`/business_transaction/productions`)
+      .all(checkJwt, checkRole(new Set([Roles.SUPERVISOR, Roles.BUSINESS])))
+      .get(async (req: express.Request, res: express.Response, next: express.NextFunction) => {
```
[the remark this answers is not in the document]
Comment (Manraj45) [reply backend/src/main/routes/TransactionRoute.ts:127]: Most of the time, in the route, you should only be calling the service methods. In rare cases, you can have logic in the route.

**Review by Manraj45 — CHANGES_REQUESTED** [review]
Need to rework the backend a bit and change the endpoints for the backend and frontend.

File: backend/src/main/routes/TransactionRoute.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (Mnx458) [reply backend/src/main/routes/TransactionRoute.ts:127]: I moved all the logic in transactionService

File: frontend/src/App.tsx [hunk 2]
```diff
@@ -89,6 +90,34 @@ const App = () => {
...
+                    if (account.authenticated) {
+                      if (
+                        account.account.role === 'ADMIN' ||
+                        account.account.role === 'SUPERVISOR' ||
+                        account.account.role === 'BUSINESS'
+                      ) {
+                        return (
+                          <React.Fragment>
+                            <Sidebar />
+                            <Financials id="1" />
```
[the remark this answers is not in the document]
Comment (Mnx458) [reply frontend/src/App.tsx]: That was a mistake

**Comment by Mnx458** [comment]
image
image

**Comment by jmezaber** [comment]
> image image

The UI is lacking a bit of colour. The mockup had colours for the table headers, it would be nice if that can be reflected in the UI. 

Also, if a project has an amount of 0$, it shouldn't appear in the table. The legend inside the pie chart is not aligned. 

I was also wondering if the page is responsive and where does the pie chart appear on the page?

File: backend/src/main/services/TransactionService.ts [hunk 3]
```diff
@@ -68,6 +73,61 @@ export class TransactionService {
...
+      projectIds.push({ projectId: project.id, name: project.title });
+    });
+    for (const project of projectIds) {
+      let wagesExpensesForProject = 0;
+      let toolsExpensesForProject = 0;
+      let othersExpensesForProject = 0;
+      const expensesOfProject = await this.expenseService.getAllExpensesForProjects(Number(project.projectId));
+      expensesOfProject?.forEach((element: any) => {
+        const expenseValue = element.dataValues.transaction.dataValues.amount;
+        console.log(element.dataValues.type);
```
Comment (nasaku898) [inline backend/src/main/services/TransactionService.ts:111]: remove console.log

Comment (Manraj45) [inline backend/src/main/services/TransactionService.ts:111]: Remove console.log

**Review by Manraj45 — CHANGES_REQUESTED** [review]
Remove console log

**Review by jmezaber — CHANGES_REQUESTED** [review]
Please bring the changes to the UI mentioned in my previous comment

**Comment by Manraj45** [comment]
> Please bring the changes to the UI mentioned in my previous comment

I implemented the changes you requested. Here is the new UI:

image
image

I also made the frontend more responsive:
image

**Comment by nasaku898** [comment]
I like the new UI changes @Manraj45 . Great work!

**Review by jmezaber — APPROVED** [review]
Thanks, @Manraj45 for the UI changes, it looks beautiful!

**Comment by huangs08** [comment]
The UI changes look great! Good job.

**Review by huangs08 — APPROVED** [review]

**Review by Manraj45 — APPROVED** [review]
Good Job @Mnx458! The backend is good now.

**Pull request merged**